### PR TITLE
Add streamdeckplus support

### PR DESCRIPTION
**Pull request by admiral0**
I cleaned up @karstlok 's pull request ( #108) to add support for the streamdeck+. Differences:

* Refactored `_read_keyl_states` into `_read_control_states`
* Renamed rotary to `dial`. It just felt wrong to me. `¯\_(ツ)_/¯`
* added new functions to check on device capabilities
* used `Enum`s for event types and control types
* moved streamdeckplus specific code to its own file
* reworked API interfaces on callbacks

```python
# no change
def callback_key(deck_instance, key_index, key_state_bool):
  pass
# changed
def callback_dial(deck_instance, dial_index, event_type: DialEventType, value):
  pass
# changed
def callback_touchscreen(deck_instance, event_type: TouchscreenEventType, value: dict):
  pass
```

### CODE REVIEW

File: src/StreamDeck/Devices/StreamDeck.py [hunk 1]
```diff
@@ -258,6 +335,25 @@ def key_image_format(self):
         .. seealso:: See :func:`~StreamDeck.set_key_image` method to update the
                      image displayed on a StreamDeck button.
 
+        :rtype: dict()
+        :return: Dictionary describing the various image parameters
+                 (size, image format).
+        """
+        return {
+            'size': (self.TOUCHSCREEN_PIXEL_WIDTH,
```
Comment (admiral0) [inline src/StreamDeck/Devices/StreamDeck.py:343]: ugh it's inverted :facepalm:

**Comment by benedikt-bartscher** [comment]
Hi @admiral0 i just tested this PR with `src/example_plus.py` and as far as i can tell everything works fine. Thank you very much!

**Comment by abcminiuser** [comment]
Thanks! I meant to get to this sooner, but I've merged a slightly tweaked version of this in 56f91960297f753acdaeb79fd460c4326f7c4840.

Unfortunately the StreamDeck + is damned expensive in Australia (and I can't justify the expense for a hobby project) and Corsair told me to shove it when I asked for even temporary access to a unit, so I'm trusting on you and the rest of the community to verify that it works before I tag off the next version for PyPi.

Can you please grab the latest master and verify the example script works on the StreamDeck +?

**Comment by admiral0** [comment]
I cannot test it for two weeks - traveling atm. Maybe @benedikt-bartscher or @joelspiers15 can.

**Comment by karstlok** [comment]
Just tested it, works.

**Comment by benedikt-bartscher** [comment]
Thanks @abcminiuser, i just tested the master branch, however the icons on the touchscreen are broken for me. They work fine in this MR. I will try to debug further later this evening.

**Comment by benedikt-bartscher** [comment]
I fixed those issues in https://github.com/abcminiuser/python-elgato-streamdeck/pull/129

**Comment by joelspiers15** [comment]
Also was seeing that same touchscreen image drawing issue, #129 fixed it.

**Pull request closed without merging**